app/src/main/java/com/infomaniak/mail/data/MailData.kt

```diff
@@ -292,6 +312,30 @@ object MailData {
         }
     }
 
+    /**
+     * Merge Realm & API data
+     */
+
+    private fun mergeContacts(realmContacts: List<Contact>, apiContacts: List<Contact>): List<Contact> {
+
+        // Get outdated data
+        Log.d("API", "Contacts: Get outdated data")
+        // val deletableContacts = ContactsController.getDeletableContact(apiContacts)
+        val deletableContacts = realmContacts.filter { realmContact ->
+            !apiContacts.any { it.id == realmContact.id }
+        }
+
+        // Save new data
+        Log.d("API", "Contacts: Save new data")
+        ContactsController.upsertContacts(apiContacts)
+
+        // Delete outdated data
+        Log.d("API", "Contacts: Delete outdated data")
+        ContactsController.deleteContacts(deletableContacts)
+
+        return apiContacts
+    }
```

Comment:
This method should rather be in `ContactsController `, since it's the same context